### ISSUE TITLE
Typing conflict in Python Series constructor

### ISSUE THREAD

**Issue by stinodego** [pola-rs/polars#960]
There's various ways to initialize a Series object in Python. This one is causing some trouble:
```
s = pl.Series([1,2,3])
```

If you type-check this with `mypy`, you'll get an error. Which makes sense, because you're passing a list to the `name` argument, which expects a string. This is enabled with a 'hack' in the code of the Series `__init__`:

```
if values is None and not isinstance(name, str):
    values = name
    name = ""
```

Unfortunately, I haven't found a great fix. I see a few ways to go about this:

1. Deprecate the syntax, and require users to type `pl.Series(values=[1,2,3])`. This is not very user-friendly.
2. Change the type of the `name` argument to also accept sequences. This means no more type errors for the case mentioned above, but then it will accept `pl.Series([1,2,3], [1,2,3])`, which we don't want.
3. Change around the argument order, so that `values` is the first argument, followed by `name`. This makes sense in a way (`pandas` does this, for example). However, this breaks a lot of stuff. Most notably:
  * `pl.Series('name', [1,2,3])` must  be replaced by `pl.Series([1,2,3], 'name')`
  * `pl.Series('name')` must be replaced by `pl.Series(name='name')`

To me, option 2 feels bad and hacky. I'd say that if there is a good reason for `name` to be the first argument, option 1 is best. If not, option 3 is best (but probably requires some work to implement).

I wonder what your views are on this!

**Comment by ritchie46** [comment]
I think option 2 is quite ok. 

In case a user does pass: `pl.Series([1,2,3], [1,2,3])` we should throw a `ValueError`, mentioning that only one sequence may be passed. I believe the types should indicate that there exists a valid permutation of the indicated types, but does not have to guarantee every permutation is valid.

And.. we don't break stuff. :)

**Comment by alippai** [comment]
Wouldn't `@overload` help here?

**Comment by stinodego** [comment]
I tried `@overload`, but as far as I've seen that doesn't work on the `__init__` of a class. Then I tried calling an overloaded function from the `__init__`, but couldn't get that to work.

Your explanation sounds reasonable @ritchie46 , I'll implement that.

**Comment by ritchie46** [comment]
Fixed in #961
